server/compile_lexicon_to_json.py

```diff
@@ -124,23 +144,127 @@ def process_row(row, json_words, json_syllables, rows_of_crossid):
+        if (row["DOCULECT"] != "Old_Burmese" and row["DOCULECT"] != "Rangoon")
+    ]
     row_tuples_sorted_by_languages = ob + rangoon + remaining
 
     # second step: sorting by senses
-    row_tuples_sorted_by_senses = sorted(row_tuples_sorted_by_languages, key = lambda x: x[1]['CONCEPT'])
+    row_tuples_sorted_by_senses = sorted(
+        row_tuples_sorted_by_languages, key=lambda x: x[1]["CONCEPT"]
+    )
     return row_tuples_sorted_by_senses
 
+
+def compile_to_json_full_cognates(path, proto, cognates="COGID"):
+    """
+    Get the JSON from a wordlist file with "normal" cognates.
+
+    Example
+    -------
+
+    compile_to_json_full_cognates(
+        "pipeline/output/germanic/stage3/germanic-aligned-final.tsv",
+        "Proto-Germanic", cognates="CROSSIDS")
+    """
+
+    data = []
+    with open(path) as f:
+        for row in f.readlines():
+            data += [[cell.strip() for cell in row.split("\t")]]
+
+    header = [row for row in data if row[0] and not row[0].startswith("#")][0]
+    data_dict = {}
+    for i, row in enumerate(data[1:]):
+        if row[0].startswith("#") or not row[0].strip():
+            pass
+        else:
+            cell_dict = dict(zip(header, row))
+            data_dict[cell_dict["ID"]] = cell_dict
+    # create the board dictionary
+    doculects = sorted(set([row["DOCULECT"] for row in data_dict.values()]))
+    boards = {
+        "fstDoculects": doculects,
+        "fstUp": {d: {} for d in doculects},
+        "fstDown": {d: {} for d in doculects},
+        "boards": {},
+        "words": {},
+        "columns": {},
+        "syllables": {},
+    }
+
+    # fill data with content by iterating over the data_dict
+    for i, row in data_dict.items():
+        idx = "word-" + str(i)
+        boards["words"][idx] = {
+            "id": idx,
+            "doculect": row["DOCULECT"],
+            "syllables": [".".join(row["TOKENS"].split())],
+            "gloss": row["CONCEPT"],
+            "glossid": row["GLOSSID"],
+        }
+        boards["syllables"][idx + "-0"] = {
+            "id": idx + "-0",
+            "doculect": row["DOCULECT"],
+            "syllables": boards["words"][idx]["syllables"],
+            "gloss": boards["words"][idx]["gloss"],
+            "glossid": boards["words"][idx]["glossid"],
+            "wordID": idx,
+            "syllOrder": 0,
+            "syllable": boards["words"][idx]["syllables"][0],
+        }
+
+        # column ids are in fact the cognate sets +++
+        cogid = "column-" + row[cognates]
+        if cogid in boards["columns"]:
+            boards["columns"][cogid]["syllableIds"] += [idx + "-0"]
+        else:
+            boards["columns"][cogid] = {"id": cogid, "syllableIds": [idx + "-0"]}
+
+        # boards are cross-semantic cognates, they group cognate sets into one
+        # group, but for normal data, this is already done, so we assign each
+        # cognate set to its own board
+        if row["DOCULECT"] == proto:
```

Comment:
Do you think we can just remove this and the subsequent code then?

Reply:
And then work on generating the boards from the transducers?

Reply:
Yes, the code with `if row["DOCULECT"]` can be removed.